### ISSUE TITLE
Fire interact issues

### ISSUE THREAD

**Issue by PEMapModder** [PocketMine/PocketMine-MP#3961]
When clicking the burning side of a block, the fire extinguishes.
This should be happening when left-clicked, not right-clicked, as in vanilla MCPE.

I located the source of this issue:
https://github.com/PocketMine/PocketMine-MP/blob/php7-0.14/src/pocketmine/level/Level.php#L1660

However I am not sure what that line was supposed to do, so I haven't removed it (yet). Putting down the issue here as a reminder to the existence of this issue.

Also, placing water on the burning side of a block empties the water bucket while extinguishing the fire. PocketMine isn't clever enough to believe that the water in the bucket evaporated because of the fire, right? :stuck_out_tongue: 

In conclusion, this issue includes three parts:
- [x] :one: Fire is not removed when left-clicking (long-clicking without necessarily removing a block) a burning side of a block.
- [x] :two: Fire is removed when right-clicking the burning side of a block.
- [x] :three: Possibly related with :two:: water bucket empties, extinguishing fire without placing water when clicking water bucket on a fire block
